src/EVM.hs

```diff
@@ -727,13 +727,23 @@ exec1 conf = do
 
         OpSload -> {-# SCC "OpSload" #-}
           case stk of
-            x:xs -> do
-              acc <- accessStorageForGas self x
-              let cost = if acc then g_warm_storage_read else g_cold_sload
-              burn cost $
-                accessStorage self x $ \y -> do
-                  next
-                  assign' (#state % #stack) (y:xs)
+            x:xs ->
+              let
+                finalizeLoad readValue = do next; assign' (#state % #stack) (readValue:xs)
+
+                symbolicRead :: EVM t s () =
+                  case this.external of
+                    True -> accessStorage self x finalizeLoad
+                    False -> finalizeLoad $ Expr.readStorage' (Expr.concKeccakOnePass x) this.storage
```

Comment:
Can we do 
```haskell
symbolicRead :: EVM t s () = if this.external
  then ...
  else ...
```

It'd be compact and clean, no?
here? It'd be one line less

Reply:
Fixed!